### DISCUSSION TITLE
Checking if latent variables correspond to covariates

### DISCUSSION THREAD

**Discussion by rokapary** [JenniNiku/gllvm#192] [Q&A]
Hello,

I hope this Q is not redundant, but I am returning to this issue after some time and seem to still be struggling with it. 
I’m exploring species co-occurrence and correlations and the effect of environmental variables on them (16 species, 1000 rows). I’ve already ran a basic GLLVM with no env. Variables included (model based on best AIC):
fit1lv <- gllvm(A, family =binomial(link="probit"), num.lv = 1, seed = 1234)

I want to see if the latent variables of this null model correspond to any of my 6 environmental covariates. These 6 covariates are already chosen from a bigger data set after removing correlated variables. 

If I understand correctly, covariates that exhibit a trend with the LVs are a good pick for the actual GLLVM model that includes env. variables as a way to ensure I am not including covariates in the model that have no bearing on the species occurrence trends. Am I correct in that assumption?

The graphs I receive look like this:
 
![Rplot](https://github.com/user-attachments/assets/77a44c6c-fbf7-436d-ad72-519da21509ed)

(for (i in 1:length(colnames(B))) {
  covariate <- B[,i]
  rbPal <- colorRampPalette(c('red', 'blue'))
  Colorsph <- rbPal(20)[as.numeric(cut(covariate, breaks = 20))]
  breaks <- seq(min(covariate), max(covariate), length.out = 30)
  ordiplot(fit1lv, main = paste0("Ordination of sites, color: ",colnames(B)[i]),
           symbols = TRUE, s.colors = Colorsph, xlim = c(-1.2,1.2), ylim = (c(-1.2, 1.2)))
})

And I am not sure how to interpret this. Is there a trend showing in any of these (red being low values and blue high values).

**Comment by BertvanderVeen** [comment]
Questions are never redundant! To me, this does not look like a pattern. However, if you want to formally assess if the latent variable depends on a covariate, I suggest you have a look at this vignette: https://jenniniku.github.io/gllvm/articles/vignette6.html#:~:text=When%20including%20predictor%20variables%2C%20the,number%20of%20parameters%20to%20estimate.

**Reply by rokapary** [reply]
Thank you! 

After reaidng through it, would it be the Coefficients LV predictors from the constrained/concurrent models that assess if the latent variable depends on the covariate?

**Reply by BertvanderVeen** [reply]
That's correct! You can find them in the summary, or with "coef".

**Reply by rokapary** [reply]
Perfect! Thank you so much!
